### PR TITLE
Backend/categories

### LINKED ISSUE #29: Implement CategoryService

### PR DESCRIPTION
**Pull request by bits0rcerer**
This PR aims to implements categories from JPA to API Controller.

Note: This PR also purges old deprecated Models, Services, Controllers, ...

Resolves #29

### CODE REVIEW

File: src/main/java/com/peerrequest/app/services/CategoryServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            return Optional.empty();
+        }
+
+        var category = optional.get();
+
+        if (newProps.name() != null) {
+            category.setName(newProps.name());
+        }
+        if (newProps.deadline() != null) {
+            category.setName(newProps.name());
```
Comment (okidokidokkaebi) [inline src/main/java/com/peerrequest/app/services/CategoryServiceImpl.java:47]: Setters are not right

File: src/main/java/com/peerrequest/app/data/Category.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    private ZonedDateTime deadline;
+    @Column(name = "min_score", nullable = false)
+    @Getter
+    @Setter
+    private float minScore;
+    @Column(name = "nax_score", nullable = false)
+    @Getter
+    @Setter
+    private float maxScore;
+
```
Comment (okidokidokkaebi) [inline src/main/java/com/peerrequest/app/data/Category.java:69]: scoreStepSize is missing

File: src/main/java/com/peerrequest/app/services/CategoryServiceImpl.java (hunk 1 above)
Comment (bits0rcerer) [reply src/main/java/com/peerrequest/app/services/CategoryServiceImpl.java:47]: nice catch

File: src/main/java/com/peerrequest/app/data/Category.java (hunk 2 above)
Comment (bits0rcerer) [reply src/main/java/com/peerrequest/app/data/Category.java:69]: yes, I do not see any value in having it. What do you think?

Comment (okidokidokkaebi) [reply src/main/java/com/peerrequest/app/data/Category.java:69]: For the different conventions of different conferences. For example, one conference may allow a rating form 0 to 1 with 0.1 steps, the next may only allow ratings from 0 to 1 with 0.25 steps.

File: src/main/java/com/peerrequest/app/api/CategoriesController.java [hunk 3]
```diff
@@ -24,80 +23,76 @@
...
-        this.mockUpCategory = category;
-        return mockUpCategory;
+        return option.get().toDto();
     }
 
-    @PatchMapping(value = "/categories/{categoryId}", consumes = "application/json", produces = "application/json")
-    public Category patchCategory(@RequestBody final Category category) {
-        return mockUpCategory;
+    @DeleteMapping("/categories/{id}")
+    Optional<Category.Dto> deleteCategory(@PathVariable Long id) {
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/CategoriesController.java:53]: check is missing if the calling user is the creator of the category when the category is external

File: src/main/java/com/peerrequest/app/services/CategoryServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        }
+        if (newProps.minScore() != null) {
+            category.setMinScore(newProps.minScore());
+        }
+
+        return Optional.of(repo.save(category));
+    }
+
+    @Override
+    public Optional<Category> delete(Long cursor) {
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/services/CategoryServiceImpl.java:63]: Needs to be implemented after other services are implemented: Logic for reassignment of the entries of the category and deletion of all corresponding entries, reviews, requests and processes.

Comment (bits0rcerer) [reply src/main/java/com/peerrequest/app/services/CategoryServiceImpl.java:63]: [Cascade Remove](https://www.baeldung.com/jpa-cascade-remove-vs-orphanremoval)

**Comment by bits0rcerer** [comment]
Note: rebased `backend/categories` onto `master`

**Review by Wil-Lee — APPROVED** [review]

**Pull request merged**